### ISSUE TITLE
[Train] `RunConfig` doesn't get propagated from the Tuner to the Trainer

### ISSUE THREAD

**Issue by justinvyu** [ray-project/ray#33539]
### What happened + What you expected to happen

When specifying `RunConfig` through the Tuner and not the Trainer, the Trainer does not have access to the run config later on.

We should either:
1. Propagate the Tuner's run config down to the trainer.
2. At least initialize the Trainer with a default run config and make it clear in the docs that you can't access these properties in a Trainer.

cc: @gjoliver 

### Versions / Dependencies

2.3.0

### Reproduction script

```python
from ray.train.torch import TorchTrainer
from ray.air.config import ScalingConfig, RunConfig
from ray.tune import Tuner


class MyTrainer(TorchTrainer):
    def training_loop(self) -> None:
        assert self.run_config.local_dir, "dude ... !"

trainer = MyTrainer(
    train_loop_per_worker=lambda: None,
    scaling_config=ScalingConfig(num_workers=2)
)

tuner = Tuner(trainer, run_config=RunConfig(local_dir="hi/jun"))

tuner.fit()
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
